### PR TITLE
Update 1

### PR DESCRIPTION
**Pull request by Git-Abhay7**

### CODE REVIEW

File: controller/userController.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors });
+    }
+    var query = { email: req.body.email };
+    const result = await user.findOne(query);
+    try {
+      if (result) {
+        if (result.email == req.body.email) {
+          res.json({
```
Comment (upexcel) [inline controller/userController.js:16]: one common function for sending error and response

File: controller/userController.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            responseCode: 409,
+            responseMessage: "Email already exist.",
+          });
+        }
+      } else {
+        const saltRounds = 10;
+        bcrypt.hash(req.body.password, saltRounds, function (err, hash) {
```
Comment (upexcel) [inline controller/userController.js:23]: read bcrypt you can use async/await

File: model/userModel.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    },
+    password: {
+      type: String,
+    },
+  },
+  {
+    timestamps: true,
+  }
+);
+module.exports = mongoose.model("user", userKey, "user");
```
Comment (upexcel) [inline model/userModel.js:22]: add all logic in models

File: routes/profileRoute.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+var router = require("express").Router();
+const { check, validationResult } = require("express-validator");
+
+const profileController = require("../controller/profileController");
+
+router.post("/profile", [check('user_id').notEmpty().withMessage("Please provide userId"),check('mobileNumber').isLength({min:10}).notEmpty().withMessage("Please provide mobileNumber")], profileController.profile);
```
Comment (upexcel) [inline routes/profileRoute.js:6]: call validators function , write validator in different folder/file

File: controller/profileController.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    try {
+      if (result) {
+        res.json({
+          responseCode: 409,
+          responseMessage: "UserId already exist.",
+        });
+      } else {
+        new profile(req.body).save( async (error, newProfile) => {
+            try{
+                await res.json({
```
Comment (saurabhketech) [inline controller/profileController.js:21]: not needed here

File: controller/profileController.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    const result = await profile.findOne({ user_id: req.body.user_id });
+
+    try {
+      if (result) {
+        res.json({
+          responseCode: 409,
+          responseMessage: "UserId already exist.",
+        });
+      } else {
+        new profile(req.body).save( async (error, newProfile) => {
```
Comment (saurabhketech) [inline controller/profileController.js:19]: use await here

File: controller/profileController.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+const profile = require("../model/usersProfile");
+const user = require("../model/userModel");
+var { check, validationResult } = require("express-validator");
+module.exports = {
+  profile: async (req, res) => {
```
Comment (saurabhketech) [inline controller/profileController.js:5]: profile: async (req, res) =>{
try{
    const errors = validationResult(req);
if (!errors.isEmpty()) {
    return   // call error handler
    }
let response = await model.createProfile(req.body)
//call success handler
}catch {
// call error handler
}

}

File: model/userModel.js (hunk 3 above)
Comment (saurabhketech) [reply model/userModel.js:22]: userModal = mongoose.model("user", userKey, "user");

userModal['createUser'] = (body)=>{

}

module.exports = useMOdel;

**Pull request merged**